tools/scripts/Makefile

```diff
@@ -372,10 +377,24 @@ post-build:
 		mv -f sw.map $(TEMP_DIR);				\
 	fi;
 
-xilinx:	xil-pre-cleanup eval-hardware pre-build xilinx-bsp
-	$(call copy_srcs,xilinx)
-	@$(MAKE) -s compile
-	rm -rf $(BUILD_DIR)/SDK.log
+# Upload the bitstream and elf file to the board
+.SILENT:run
+run: eval-hardware
```

Comment:
`run` should depend on the source compilation rule, otherwise it might upload an old binary

Reply:
Doing this means that the user can run the binary only once per build.